src/openms/source/ANALYSIS/ID/FragmentIndex.cpp

```diff
@@ -0,0 +1,650 @@
+          init_hits.hits_.resize(0);
+      }
+
+
+      for (auto hit_iter = init_hits.hits_.rbegin(); hit_iter != init_hits.hits_.rend(); ++hit_iter)
+      {
+        if (hit_iter->num_matched_ >= min_matched_peaks_)           // search for the first element that should be included
+        {
+          init_hits.hits_.resize(init_hits.hits_.size() - (distance(init_hits.hits_.rbegin(), hit_iter)));
+          break;
+        }
+      }
+      /* alternative code
+       * auto it_zero = std::lower_bound(init_hits.hits_.begin(), init_hits.hits_.end(), min_matched_peaks_ , [](const SpectrumMatch& sm, uint32_t b){
+return sm.num_matched_ > b;
+});
+
+if (it_zero != init_hits.hits_.end() && it_zero->num_matched_ == 0)
+{
+init_hits.hits_.erase(it_zero, init_hits.hits_.end());
+}
+       * */
+
+  }
+
+  void FragmentIndex::searchDifferentPrecursorRanges(const MSSpectrum& spectrum,
+                                                     float precursor_mass,
+                                                     SpectrumMatchesTopN& sms,
+                                                     uint16_t charge)
+  {
+      int16_t  min_isotope_error_applied;
+      int16_t  max_isotope_error_applied;
+      float precursor_window_upper_applied;
+      float precursor_window_lower_applied;
+      if (open_search_)
+      {
+        min_isotope_error_applied = 0;
+        max_isotope_error_applied = 0;
+        precursor_window_upper_applied = open_precursor_window_upper_;
+        precursor_window_lower_applied = open_precursor_window_lower_;
+      }
+      else
+      {
+        min_isotope_error_applied = min_isotope_error_;
+        max_isotope_error_applied = max_isotope_error_;
+        precursor_window_upper_applied = 0;
+        precursor_window_lower_applied = 0;
+      }
+
+      for (int16_t isotope_error = min_isotope_error_applied; isotope_error <= max_isotope_error_applied; isotope_error++)
+      {
+        SpectrumMatchesTopN candidates_iso_error;
+        float precursor_mass_isotope_error = precursor_mass + ((float)isotope_error * (float)Constants::C13C12_MASSDIFF_U);
+        auto candidates_range = getPeptidesInPrecursorRange(precursor_mass_isotope_error, {precursor_window_lower_applied, precursor_window_upper_applied}); // for the simple search we do not apply any modification window!!
+        candidates_iso_error.hits_.resize(candidates_range.second - candidates_range.first + 1);
+
+        queryPeaks(candidates_iso_error, spectrum, candidates_range, isotope_error, charge);
+
+        // take only top 50 hits
+        //trimHits(candidates_iso_error);
+
+        sms += candidates_iso_error;
+      }
+      //trimHits(sms);
+  }
+
+  void FragmentIndex::querySpectrum(const OpenMS::MSSpectrum& spectrum,
+                                    OpenMS::FragmentIndex::SpectrumMatchesTopN& sms)
+  {
+      if (!isBuild())
+      {
+        OPENMS_LOG_WARN << "FragmentIndex not yet build \n";
+        return;
```

Comment:
this would warrant a `throw`, but since building will be part of Ctor, this code is void anyways?